### PR TITLE
Update ENLSVG names

### LINKED ISSUE #2708: Update Enlsvg to follow codebase naming guidelines
### Description of the task

We have a software guideline that treats all acronyms as words. For example, we would treat the word "GUI" as "Gui" when we write C++ code. In our codebase, our motion planner for navigating around static obstacles like the Defence Area (called ENLSVG) often breaks this guideline because we took this from an external repo. We want to integrate this into our codebase so we can maintain our own version locally.

### Acceptance criteria

- [ ] Navigate to the `Software` folder in the terminal; if you have `ag-silversearcher ` installed, you can run `ag ENLSVG` to find all the instances of the acronym ENLSVG in our codebase. Alternatively you can go to the `Software` folder and run `grep -nr "ENLSVG"`. This looks into every file and finds the word "ENLSVG" inside files. If you're using CLion, you can use `ctrl + shift + f` to search for word in the entire project as well.
- [ ] For each instance of "ENLSVG" you find above, replace it with "Enlsvg" to follow our code naming guidelines. Change it if its the name of a file, but it's not necessary to change it if it's in a comment
- [ ] Fix any build issues that come up when running `bazel build //software/...` in the Software/src directory

### PR DESCRIPTION
**Pull request by raymond212**
### Description

Replaced all instances of the acronym "ENLSVG" with "Enlsvg" to follow codebase naming guidelines. 
### Testing Done

Successfuly builds and passes all tests. 
### Resolved Issues

Resolves #2708. 
### Review Checklist

**_It is the reviewers responsibility to also make sure every item here has been covered_**

- [ ] **Function & Class comments**: All function definitions (usually in the `.h` file) should have a javadoc style comment at the start of them. For examples, see the functions defined in `thunderbots/software/geom`. Similarly, all classes should have an associated Javadoc comment explaining the purpose of the class.
- [ ] **Remove all commented out code**
- [ ] **Remove extra print statements**: for example, those just used for testing
- [ ] **Resolve all TODO's**: All `TODO` (or similar) statements should either be completed or associated with a github issue

### CODE REVIEW

**Review by LiCody — DISMISSED** [review]
LGTM

**Review by nimazareian — CHANGES_REQUESTED** [review]
Overall looks good to me. I noticed that the file names were also updated. In our code base, unlike the class names, we use snake_case for file names. Could you update the file names under `src/extlibs/enlsvg` to follow that guide line instead?

**Comment by raymond212** [comment]
> Overall looks good to me. I noticed that the file names were also updated. In our code base, unlike the class names, we use snake_case for file names. Could you update the file names under `src/extlibs/enlsvg` to follow that guide line instead?

I renamed all of the files in `src/extlibs/enlsvg` to use snake_case. Everything builds successfully and all tests pass.

**Review by nimazareian — APPROVED** [review]
LGTM! 💯

**Review by LiCody — APPROVED** [review]

File: src/extlibs/enlsvg/README.md [hunk 1]
```diff
@@ -1,7 +1,7 @@
-# ENLSVG-Pathfinding
+# Enlsvg-Pathfinding
 This is an implementation of the Edge N-Level Sparse Visibility Graph algorithm for extremely fast optimal Any-Angle Pathfinding on grid maps. (on the order of a millisecond per path computation on 6000x6000 grids)
 
-The ENLSVG algorithm requires a pre-processing step, which may take some time, depending on the size of the grid map.
+The Enlsvg algorithm requires a pre-processing step, which may take some time, depending on the size of the grid map.
```
Comment (germchow) [inline src/extlibs/enlsvg/README.md:4]: I think we can leave it as ENLSVG (all caps) in the readme? line 1 as well

Comment (raymond212) [reply src/extlibs/enlsvg/README.md:4]: Oh, thanks for pointing that out! I created a new pull request #2755 to fix that.

**Pull request merged**